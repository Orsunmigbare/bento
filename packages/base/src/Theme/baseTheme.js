import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
