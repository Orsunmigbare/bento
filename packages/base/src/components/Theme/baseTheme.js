export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
