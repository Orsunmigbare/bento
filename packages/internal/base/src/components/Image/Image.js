import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { themify, themeProps } from "@staccx/theme"

const Image = ({
  src,
  className,
  round,
  variant,
  aspectRatio,
  stretch,
  ...otherProps
}) => {
  if (aspectRatio) {
    const antecedent = parseInt(aspectRatio.split(":")[0], 10)
    const consequent = parseInt(aspectRatio.split(":")[1], 10)
    const ratio = consequent / antecedent
    console.log("Image:", consequent, antecedent, ratio)
    return (
      <CropContainer
        className={className}
        ratio={ratio}
        variant={variant}
        round={round}
        {...otherProps}
      >
        <CroppedImg src={src} stretch={stretch} />
      </CropContainer>
    )
  }
  return (
    <Img
      src={src}
      className={className}
      round={round}
      variant={variant}
      {...otherProps}
    />
  )
}

export const IMAGE = "IMAGE"
export const IMAGE_CONTAINER = "IMAGE_CONTAINER"

export const Img = styled.img`
  ${p => p.round && "border-radius: 50%"};
  ${themify(IMAGE)};
`

export const CroppedImg = styled(Img)`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  max-width: none;
  min-height: 100%;
  min-width: 100%;
`

export const CropContainer = styled.div`
  position: relative;
  display: block;
  overflow: hidden;
  padding-bottom: ${p => p.ratio * 100}%;
  ${p => p.round && "border-radius: 50%"};
  ${themify(IMAGE_CONTAINER)};
`

Image.defaultProps = {
  className: "",
  size: "medium"
}

Image.propTypes = {
  children: themeProps.children,
  className: PropTypes.string,
  size: themeProps.spacing
}

export default Image
