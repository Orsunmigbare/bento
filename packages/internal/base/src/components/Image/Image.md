```jsx
<div>
  <Image src="http://via.placeholder.com/150x150" />
  <Image src="http://via.placeholder.com/150x150" round />
  <Image src="http://via.placeholder.com/250x150" round aspectRatio={"1:1"} />
  <Image src="http://via.placeholder.com/250x150" aspectRatio={"16:9"} />
</div>
```
