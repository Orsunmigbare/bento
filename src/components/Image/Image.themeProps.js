import { themePropTypes } from "../../theming"

const themeProps = {
  image: {
    name: "IMAGE",
    description: "image style",
    type: themePropTypes.style
  },
  container: {
    name: "IMAGE_CONTAINER",
    description: "image container style",
    type: themePropTypes.style
  }
}

export default themeProps
