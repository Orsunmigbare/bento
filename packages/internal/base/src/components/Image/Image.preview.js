import React from "react"
import Image from "./Image"
import Layout from "../Layout/Layout/Layout"
import Label from "../Forms/Label/Label"
import Toggle from "../Forms/Toggle/Toggle"
import Slider from "../Forms/Slider/Slider/Slider"
import Input from "../Forms/Input/Input"

const preview = {
  title: "Image",
  category: "Components/Image",
  tags: ["image"],
  component: Image,
  render: props => (
    <Image src="http://fillmurray.com/280/200" aspectRatio={"1:1"} {...props} />
  ),
  renderExampleController: ({ setComponentState }) => (
    <Layout grid="columns">
      <div>
        <Label for="roundRadio">round</Label>
        <Toggle
          id="roundRadio"
          onChange={e => setComponentState({ round: e.target.checked })}
        >
          round
        </Toggle>
      </div>
      <div>
        <Label>Width</Label>
        <Slider
          name={"WidthExample"}
          min={100}
          max={300}
          step={10}
          onChange={e =>
            setComponentState({ width: parseInt(e.target.value, 10) })
          }
        />
      </div>
      <div>
        <Input
          id="dsag43356"
          label="AspectRatio"
          onChange={e => setComponentState({ aspectRatio: e.target.value })}
          placeholder={"1:1"}
          type="text"
        />
      </div>
    </Layout>
  )
}

export default preview
