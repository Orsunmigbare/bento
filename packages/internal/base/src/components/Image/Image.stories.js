import React from "react"
import { storiesOf } from "@storybook/react"
import { withKnobs, number, select } from "@storybook/addon-knobs"
import docs from "./Image.md"
import Image from "./Image"

storiesOf("components|base/Image", module)
  .addDecorator(withKnobs)
  .add(
    "Image",
    () => {
      const label = "Width"
      const defaultValue = "1"
      const options = {
        min: 1,
        max: 300,
        step: 5
      }
      const value = number(label, defaultValue, options)
      return (
        <Image
          src="https://www.w3schools.com/w3css/img_snowtops.jpg"
          width={value}
        />
      )
    },
    {
      info: {
        text: docs
      }
    }
  )
  .add(
    "Round",
    () => {
      const label = "Width"
      const defaultValue = "1"
      const options = {
        min: 1,
        max: 300,
        step: 5
      }
      const value = number(label, defaultValue, options)
      return (
        <Image
          src="https://www.w3schools.com/w3css/img_snowtops.jpg"
          width={value}
          round
        />
      )
    },
    {
      info: {
        text: docs
      }
    }
  )
  .add(
    "Img size small",
    () => {
      const label = "St√∏rrelse"
      const options = {
        huge: "huge",
        large: "large",
        mediumPlus: "mediumPlus",
        medium: "medium",
        small: "small",
        tiny: "tiny",
        micro: "micro",
        flush: "flush"
      }
      const defaultValue = "small"
      const value = select(label, options, defaultValue)
      return (
        <Image
          src="https://www.w3schools.com/w3css/img_snowtops.jpg"
          size={value}
        />
      )
    },
    {
      info: {
        text: docs
      }
    }
  )
