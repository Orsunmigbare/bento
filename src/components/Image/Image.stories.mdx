import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks"
import Image from "./Image"

<Meta title="Components|Image" component={Image} />

# Image
This is the Image component. Remember to use the src prop!


```javascript
import { Image } from "@staccx/bento"
```

There are several ways to display an image with the Image component: round, aspectRatio, size, and children.

## Standard Image

<Preview>
  <Story name="Standard">
    <Image src="http://via.placeholder.com/150x150"/>
  </Story>
</Preview>

## Rounded Image
The rounded image uses the prop round and is a boolean.

<Preview>
  <Story name="Rounded">
    <Image src="http://via.placeholder.com/150x150" round/>
  </Story>
</Preview>

## Aspect ratio
You can use the aspect ratio that takes a String.

<Preview>
    <Story name="Aspect ratio">
        <Image src="http://via.placeholder.com/150x150" aspectRatio={"16:9"}/>
    </Story>
</Preview>



## Props for Image

<Props of={Image} />
