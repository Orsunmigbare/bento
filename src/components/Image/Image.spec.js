import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../theming/themes/baseTheme"
import Image from "./Image"

describe("Image", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Image />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("ClassName", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Image className="Name" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("size", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Image size="huge" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("invalid size", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Image size={100} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("round", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Image round />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("aspectRatio", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Image aspectRatio="16:9" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("all", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Image className="Name" size="huge" round aspectRatio="16:9" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
