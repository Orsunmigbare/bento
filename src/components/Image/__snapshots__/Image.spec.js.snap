// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Snapshots ClassName 1`] = `
<img
  className="sc-AykKC pywwI Name"
  size="medium"
/>
`;

exports[`Image Snapshots No props 1`] = `
<img
  className="sc-AykKC pywwI"
  size="medium"
/>
`;

exports[`Image Snapshots all 1`] = `
<div
  className="sc-AykKE jrtjkV Name"
  size="huge"
>
  <img
    className="sc-AykKC sc-AykKD fKziJC"
  />
</div>
`;

exports[`Image Snapshots aspectRatio 1`] = `
<div
  className="sc-AykKE clwzYy"
  size="medium"
>
  <img
    className="sc-AykKC sc-AykKD fKziJC"
  />
</div>
`;

exports[`Image Snapshots invalid size 1`] = `
<img
  className="sc-AykKC pywwI"
  size={100}
/>
`;

exports[`Image Snapshots round 1`] = `
<img
  className="sc-AykKC igjITT"
  size="medium"
/>
`;

exports[`Image Snapshots size 1`] = `
<img
  className="sc-AykKC pywwI"
  size="huge"
/>
`;
